EuroCarsUSA/Views/Admin/EditCatalog.cshtml

```diff
@@ -1 +1,112 @@
+    <input type="hidden" id="deletedCarsIds" name="deletedCarsIds" value="" />
+    <table>
+        <thead>
+            <tr>
+                <th>Make</th>
+                <th>Type</th>
+                <th>Model</th>
+                <th>Color</th>
+                <th>VIN</th>
+                <th>Images</th>
+                <th>Mileage</th>
+                <th>Year</th>
+                <th>Fuel Type</th>
+                <th>Engine Volume</th>
+                <th>Transmission</th>
+                <th>Price</th>
+                <th>Actions</th>
+            </tr>
+        </thead>
+        <tbody>
+            @for (int i = 0; i < Model.Count(); i++)
+            {
+                var car = Model.ElementAt(i);
+                var uniqueId = $"car-{i}";
+                <tr id="car-row-@uniqueId">
+                    <td><component type="typeof(EditableField<Guid>)" render-mode="ServerPrerendered" param-Value="car.Id" param-Name="@($"[{i}].Id")" param-IsVisible="false" /></td>
+                    <td><component type="typeof(EnumDropdown<CarMake>)" render-mode="ServerPrerendered" param-SelectedValue="car.Make" param-SelectedValueChanged="@EventCallback.Factory.Create<CarMake>(this, value => car.Make = value)" param-Name="@($"[{i}].Make")" /></td>
+                    <td><component type="typeof(EnumDropdown<CarType>)" render-mode="ServerPrerendered" param-SelectedValue="car.Type" param-SelectedValueChanged="@EventCallback.Factory.Create<CarType>(this, value => car.Type = value)" param-Name="@($"[{i}].Type")" /></td>
+                    <td><component type="typeof(EditableField<string>)" render-mode="ServerPrerendered" param-Value="car.Model" param-ValueChanged="@EventCallback.Factory.Create<string>(this, value => car.Model = value)" param-Name="@($"[{i}].Model")" /></td>
+                    <td><component type="typeof(EnumDropdown<CarColor>)" render-mode="ServerPrerendered" param-SelectedValue="car.Color" param-SelectedValueChanged="@EventCallback.Factory.Create<CarColor>(this, value => car.Color = value)" param-Name="@($"[{i}].Color")" /></td>
+                    <td>
+                        <partial name="~/Views/Shared/Components/ImageEditor.cshtml" model="Tuple.Create(car.Images, uniqueId)" />
+                        <input type="hidden" id="ImagesJson-@uniqueId" name="@($"[{i}].ImagesJson")" value="@car.ImagesJson" />
+                    </td>
+                    <td><component type="typeof(EditableField<string>)" render-mode="ServerPrerendered" param-Value="car.VIN" param-ValueChanged="@EventCallback.Factory.Create<string>(this, value => car.VIN = value)" param-Name="@($"[{i}].VIN")" /></td>
+                    <td><component type="typeof(EditableField<string>)" render-mode="ServerPrerendered" param-Value="car.Mileage.ToString()" param-ValueChanged="@EventCallback.Factory.Create<string>(this, value => car.Mileage = int.Parse(value))" param-Name="@($"[{i}].Mileage")" /></td>
```

Comment:
For LineEdit component in year and milage should be avaiability to enter only numbers. Now I in field "Millage" i can type "some text in field for only numbers".